common/msg.c

```diff
@@ -713,13 +713,18 @@ void mp_msg_update_msglevels(struct mpv_global *global, struct MPOpts *opts)
     }
 }
 
-void mp_msg_force_stderr(struct mpv_global *global, bool force_stderr)
+bool mp_msg_force_stderr(struct mpv_global *global, bool force_stderr)
 {
     struct mp_log_root *root = global->log->root;
 
     pthread_mutex_lock(&root->lock);
     root->force_stderr = force_stderr;
     pthread_mutex_unlock(&root->lock);
+    if (force_stderr) {
+        if (fseek(stdout, 0, SEEK_SET))
```

Comment:
This would not work when stdout is a pipe.

Also causes a null-deref in failure case:

```
../common/encode_lavc.c:232:9: runtime error: member access within null pointer of type 'struct AVFormatContext'
AddressSanitizer:DEADLYSIGNAL
=================================================================
==28820==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000020 (pc 0x7fef8e5f930b bp 0x000000000020 sp 0x7ffd962c8ea0 T0)
==28820==The signal is caused by a READ memory access.
==28820==Hint: address points to the zero page.
    #0 0x7fef8e5f930b in avio_closep (/usr/lib64/libavformat.so.58+0x6e30b)
    #1 0x490168 in encode_lavc_free ../common/encode_lavc.c:232
    #2 0x48e833 in encode_lavc_init ../common/encode_lavc.c:169
    #3 0x7e8f24 in mp_initialize ../player/main.c:404
    #4 0x7e971f in mpv_main ../player/main.c:436
    #5 0xa69f2d in main ../osdep/main-fn-unix.c:5
```

To reproduce:

```console
$ mpv --of=matroska --o=- "somevid.mp4" | tee /tmp/out.mkv >/dev/null
```

Reply:
Ah dang good point there. Hmm.